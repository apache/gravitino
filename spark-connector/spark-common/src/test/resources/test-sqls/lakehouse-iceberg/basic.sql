CREATE DATABASE IF NOT EXISTS mydatabase;
USE mydatabase;

DROP TABLE IF EXISTS employee;

CREATE TABLE IF NOT EXISTS employee (
  id bigint,
  name string,
  department string,
  hire_date timestamp
) USING iceberg
PARTITIONED BY (days(hire_date));

INSERT INTO employee
VALUES
(1, 'Alice', 'Engineering', TIMESTAMP '2021-01-01 09:00:00'),
(2, 'Bob', 'Marketing', TIMESTAMP '2021-02-01 10:30:00'),
(3, 'Charlie', 'Sales', TIMESTAMP '2021-03-01 08:45:00');

SELECT * FROM employee WHERE date(hire_date) = '2021-01-01';

UPDATE employee SET department = 'Jenny' WHERE id = 1;

DELETE FROM employee WHERE id < 2;

MERGE INTO employee
USING (SELECT 4 as id, 'David' as name, 'Engineering' as department, TIMESTAMP '2021-04-01 09:00:00' as hire_date) as new_employee
ON employee.id = new_employee.id
WHEN MATCHED THEN UPDATE SET *
WHEN NOT MATCHED THEN INSERT *;

MERGE INTO employee
USING (SELECT 4 as id, 'David' as name, 'Engineering' as department, TIMESTAMP '2021-04-01 09:00:00' as hire_date) as new_employee
ON employee.id = new_employee.id
WHEN MATCHED THEN DELETE
WHEN NOT MATCHED THEN INSERT *;

SELECT * FROM employee;

