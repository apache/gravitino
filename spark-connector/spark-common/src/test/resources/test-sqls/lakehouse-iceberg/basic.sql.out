-- Automatically generated by Gravitino Spark SQL test
-- !query
CREATE DATABASE IF NOT EXISTS mydatabase
-- !query schema
struct<>
-- !query output



-- !query
USE mydatabase
-- !query schema
struct<>
-- !query output



-- !query
DROP TABLE IF EXISTS employee
-- !query schema
struct<>
-- !query output



-- !query
CREATE TABLE IF NOT EXISTS employee (
  id bigint,
  name string,
  department string,
  hire_date timestamp
) USING iceberg
PARTITIONED BY (days(hire_date))
-- !query schema
struct<>
-- !query output



-- !query
INSERT INTO employee
VALUES
(1, 'Alice', 'Engineering', TIMESTAMP '2021-01-01 09:00:00'),
(2, 'Bob', 'Marketing', TIMESTAMP '2021-02-01 10:30:00'),
(3, 'Charlie', 'Sales', TIMESTAMP '2021-03-01 08:45:00')
-- !query schema
struct<>
-- !query output



-- !query
SELECT * FROM employee WHERE date(hire_date) = '2021-01-01'
-- !query schema
struct<id:bigint,name:string,department:string,hire_date:timestamp>
-- !query output
1	Alice	Engineering	2021-01-01 09:00:00


-- !query
UPDATE employee SET department = 'Jenny' WHERE id = 1
-- !query schema
struct<>
-- !query output



-- !query
DELETE FROM employee WHERE id < 2
-- !query schema
struct<>
-- !query output



-- !query
MERGE INTO employee
USING (SELECT 4 as id, 'David' as name, 'Engineering' as department, TIMESTAMP '2021-04-01 09:00:00' as hire_date) as new_employee
ON employee.id = new_employee.id
WHEN MATCHED THEN UPDATE SET *
WHEN NOT MATCHED THEN INSERT *
-- !query schema
struct<>
-- !query output



-- !query
MERGE INTO employee
USING (SELECT 4 as id, 'David' as name, 'Engineering' as department, TIMESTAMP '2021-04-01 09:00:00' as hire_date) as new_employee
ON employee.id = new_employee.id
WHEN MATCHED THEN DELETE
WHEN NOT MATCHED THEN INSERT *
-- !query schema
struct<>
-- !query output



-- !query
SELECT * FROM employee
-- !query schema
struct<id:bigint,name:string,department:string,hire_date:timestamp>
-- !query output
2	Bob	Marketing	2021-02-01 10:30:00
3	Charlie	Sales	2021-03-01 08:45:00
